### ISSUE TITLE
run_glue : Evaluating in every grad_accumulation_step if flag eval during training is true

### ISSUE THREAD

**Issue by pratyay-banerjee** [huggingface/pytorch-transformers#935]
https://github.com/huggingface/pytorch-transformers/blob/f2a3eb987e1fc2c85320fc3849c67811f5736b50/examples/run_glue.py#L154

**Comment by pratyay-banerjee** [comment]
if (step + 1) % args.gradient_accumulation_steps == 0:
                scheduler.step()  # Update learning rate schedule
                optimizer.step()
                model.zero_grad()
                global_step += 1

                if args.local_rank in [-1, 0] and args.logging_steps > 0 and global_step % args.logging_steps == 0:
                    # Log metrics
                    if args.local_rank == -1 and args.evaluate_during_training:  # Only evaluate when single GPU otherwise metrics may not average well
                        results = evaluate(args, model, tokenizer)
                        for key, value in results.items():
                            tb_writer.add_scalar('eval_{}'.format(key), value, global_step)`
`

**Comment by pratyay-banerjee** [comment]
Found the fix.
